### PR TITLE
MES-1827 display personal commitments

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description and relevant Jira numbers
- Creating `PersonalCommitment` component to display DE personal commitments on the journal

Screenshot 2019-08-12 at 15 45 00
Screenshot 2019-08-12 at 15 45 08

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/personal-commitment/personal-commitment.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+personal-commitment {
+  ion-card {
+      border: 1px solid map-get($colors-gds, 'gds-grey-2');
+
+    .slot-row {
+      padding: 0.5em 0;
+      height: 50px;
+      background: map-get($colors-gds, 'gds-grey-5');
+      // border: 1px solid map-get($colors-gds, 'gds-grey-2');
```
Comment (TomBillingtonUK) [inline src/pages/journal/personal-commitment/personal-commitment.scss]: comment

File: src/pages/journal/journal.ts [hunk 2]
```diff
@@ -216,6 +217,12 @@ export class JournalPage extends BasePageComponent implements OnInit {
       (<SlotComponent>componentRef.instance).hasSlotChanged = slot.hasSlotChanged;
       (<SlotComponent>componentRef.instance).showLocation = (slot.slotData.testCentre.centreName !== lastLocation);
       lastLocation = slot.slotData.testCentre.centreName;
+
+      // if this is a personal commitment assign it to the component
+      if (slot.component.name === PersonalCommitmentSlotComponent.name) {
+        (<PersonalCommitmentSlotComponent>componentRef.instance).personalCommitments =
```
Comment (TomBillingtonUK) [inline src/pages/journal/journal.ts]: Can we not assign this no matter the slot type as it's on the SlotComponent interface. It just won't be used on the other ones

Comment (MarkPatt89) [reply src/pages/journal/journal.ts]: Yes a good point. I'll take a look now.

File: src/pages/journal/personal-commitment/personal-commitment.scss (hunk 1 above)
Comment (MarkPatt89) [reply src/pages/journal/personal-commitment/personal-commitment.scss]: :@

**Review by TomBillingtonUK — APPROVED** [review]

**Review by roberthall12 — APPROVED** [review]
Nice one

**Pull request merged**